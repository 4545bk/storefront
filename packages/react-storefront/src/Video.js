/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import { Helmet } from 'react-helmet'
import withStyles from '@material-ui/core/styles/withStyles'
import { inject } from 'mobx-react'

/**
 * AMP ready video component
 *
 * Usage:
 *
 * <Video
 *  src="https://amp.dev/static/inline-examples/videos/kitten-playing.webm"
 *  controls
 * />
 *
 */
export const styles = theme => ({
  root: {
    height: '100%',
    width: '100%',
    overflow: 'hidden'
  }
})

@withStyles(styles, { name: 'RSFVideo' })
@inject(({ app }) => ({ amp: app.amp }))
export default class Video extends Component {
  static defaultProps = {
    controls: true
  }
  render() {
    const { children, classes, amp, ...props } = this.props
    const Tag = amp ? 'amp-video' : 'video'
    const tagProps = {
      [amp ? 'class' : 'className']: classes.root,
      ...props
    }
    // Layout must be declared in AMP, so we use "fill" by default if not defined
    if (amp && !props.height && !props.width && !props.layout) {
      tagProps.layout = 'fill'
    }
    return (
      <React.Fragment>
        {amp && (
          <Helmet>
            <script
              async
              custom-element="amp-video"
              src="https://cdn.ampproject.org/v0/amp-video-0.1.js"
            />
          </Helmet>
        )}
        <Tag {...tagProps}>{children}</Tag>
      </React.Fragment>
    )
  }
}
