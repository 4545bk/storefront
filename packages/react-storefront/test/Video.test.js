/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Video from '../src/Video'
import Provider from './TestProvider'

describe('Video', () => {
  it('should render with video tag when in non-AMP mode', () => {
    expect(
      mount(
        <Provider app={{ amp: false }}>
          <Video src="haha.mp4" />
        </Provider>
      )
    ).toMatchSnapshot()
  })
  it('should render with amp-video in amp mode', () => {
    const wrapper = mount(
      <Provider app={{ amp: true }}>
        <Video src="haha.mp4" />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should use layout fill when not defined in amp mode', () => {
    const wrapper = mount(
      <Provider app={{ amp: true }}>
        <Video src="haha.mp4" />
      </Provider>
    )
    expect(wrapper.find('amp-video').prop('layout')).toBe('fill')
  })
  it('should use controls=true by default', () => {
    const wrapper = mount(
      <Provider app={{ amp: false }}>
        <Video src="haha.mp4" />
      </Provider>
    )
    expect(wrapper.find('video').prop('controls')).toBe(true)
  })
  it('should use class prop when in AMP mode', () => {
    const wrapper = mount(
      <Provider app={{ amp: true }}>
        <Video src="haha.mp4" />
      </Provider>
    )
    expect(wrapper.find('amp-video').props()).toHaveProperty('class')
  })
})
