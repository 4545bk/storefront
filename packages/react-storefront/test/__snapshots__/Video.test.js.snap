// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video should render with amp-video in amp mode 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Video) src="haha.mp4">
        <inject-Video src="haha.mp4" classes={{...}}>
          <Video src="haha.mp4" classes={{...}} amp={true} controls={true}>
            <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
              <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <amp-video class="RSFVideo-root-2" src="haha.mp4" controls={true} layout="fill" />
          </Video>
        </inject-Video>
      </WithStyles(inject-Video)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`Video should render with video tag when in non-AMP mode 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Video) src="haha.mp4">
        <inject-Video src="haha.mp4" classes={{...}}>
          <Video src="haha.mp4" classes={{...}} amp={false} controls={true}>
            <video className="RSFVideo-root-1" src="haha.mp4" controls={true} />
          </Video>
        </inject-Video>
      </WithStyles(inject-Video)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
